sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/multi_execution_aggregator.py

```diff
@@ -129,6 +136,37 @@ def fetch_next_block(self):
 
         raise NotImplementedError("You should use pipeline's fetch_next_block.")
 
+    def _repair_document_producer(self):
+        # refresh the routing provider to get the newly initialized one post-refresh
+        self._routing_provider = self._client._routing_map_provider
+        # will be a list of (partition_min, partition_max) tuples
+        targetPartitionRanges = self._get_target_partition_key_range()
+
+        targetPartitionQueryExecutionContextList = []
+        for partitionTargetRange in targetPartitionRanges:
+            # create and add the child execution context for the target range
+            targetPartitionQueryExecutionContextList.append(
+                self._createTargetPartitionQueryExecutionContext(partitionTargetRange)
+            )
+
+        self._orderByPQ = _MultiExecutionContextAggregator.PriorityQueue()
+
+        for targetQueryExContext in targetPartitionQueryExecutionContextList:
+            try:
+                # TODO: we can also use more_itertools.peekable to be more python friendly
+                targetQueryExContext.peek()
+                # if there are matching results in the target ex range add it to the priority queue
+
+                self._orderByPQ.push(targetQueryExContext)
+
+            except exceptions.CosmosHttpResponseError as e:
+                if document_producer.partition_range_is_gone(e):
+                    # error shouldn't bubble up all the way up here, but we raise to make sure we retry if it does
```

Comment:
in case the exception reached here, where we are going to handle the exception?

Reply:
it didn't at any point honestly, I can probably take out this logic now that I think about it

Reply:
will this also handle multiple split (split after split? ) scenario?

Reply:
Discussed offline, Ideally, the code should handle multiple splits scenario. But generally split will take some time, the one split right after another one is rare to happen, so will not put too much effort it at this moment.